### ISSUE TITLE
Feature Request: Exclusion of emoji for the CSS rules doing gradient text coloring

### ISSUE THREAD

**Issue by Hunter-Bobeck** [caro401/royal-velvet#11]
**Feature Requested**
Exclusion of emoji for the CSS rules doing gradient text coloring: whenever there is a gradient over text such as headers and file titles, the emoji is painted over by this, making emoji hard to distinguish because they lose their color. An option to show original emoji colors would be very helpful and make this theme more worth using. This might be solved be a CSS or JS solution. This is considered here, for instance: https://css-tricks.com/excluding-emojis-from-transparent-text-clipping/
A Google search yields a possible solution:
https://stackoverflow.com/questions/68731016/css-background-text-clipping-affecting-emoji

**Relevant Screenshots**
![image](https://user-images.githubusercontent.com/8333904/162130268-8e98edb2-6eeb-4c23-9445-46c08098979e.png)

**Checklist**
- [x] The feature would be useful to more users than just me. (→ If it's only relevant to you, have you considered using a [CSS Snippet](https://help.obsidian.md/How+to/Add+custom+styles.md#Use+Themes+and/or+CSS+snippets)?)

**Comment by caro401** [comment]
These solutions all need some changes in the markup so that emoji can be identified as different from normal letters. As far as I know, CSS alone can't do that, and Obsidian's markup doesn't distinguish emoji. I'll put together a plugin that will wrap emoji in a span with a specific class, so the theme can target the emoji that way and reset the text colour

**Comment by caro401** [comment]
There's a workaround using "Dynamic Highlights" plugin, and importing the code below:
```
{
  "emoji": {
    "class": "emoji",
    "color": "",
    "regex": true,
    "query": "\\p{Extended_Pictographic}",
    "mark": [
      "match",
      "start",
      "end"
    ],
    "css": ".emoji { color:black;  -webkit-text-fill-color: black; }"
  }
}
```

**Comment by Hunter-Bobeck** [comment]
> There's a workaround using "Dynamic Highlights" plugin, and importing the code below:
> 
> ```
> {
>   "emoji": {
>     "class": "emoji",
>     "color": "",
>     "regex": true,
>     "query": "\\p{Extended_Pictographic}",
>     "mark": [
>       "match",
>       "start",
>       "end"
>     ],
>     "css": ".emoji { color:black;  -webkit-text-fill-color: black; }"
>   }
> }
> ```

This helps... it doesn't work for the note titles, but it does work for the note headers.

**Comment by caro401** [comment]
Yeah it doesn't work in reading mode either :( I'll make an update shortly to take the gradient off note titles, because I don't think there's a way to target them with a plugin, and work on a plugin solution when I have a bit of free time

**Comment by caro401** [comment]
I've had a crack at a plugin for reader mode, available [here](https://github.com/caro401/obsidian-emoji-styling) - would you be comfortable doing a manual install of the plugin to see if it works for you and solves the problem?

**Comment by Hunter-Bobeck** [comment]
> I've had a crack at a plugin for reader mode, available [here](https://github.com/caro401/obsidian-emoji-styling) - would you be comfortable doing a manual install of the plugin to see if it works for you and solves the problem?

I just installed it. I'm experiencing this:
editing mode:
![image](https://user-images.githubusercontent.com/8333904/165629952-8259318d-f8ee-426a-80f6-b24f4472b126.png)
reading mode:
![image](https://user-images.githubusercontent.com/8333904/165629984-4e6608ad-9019-41cc-a348-33fd1fe69e31.png)

It seems to be having an issue in reading mode.

**Comment by caro401** [comment]
Are you using a plugin to make the emoji be images? It looks like your markup is different to what I see in my obsidian, so the plugin doesn't handle it

**Comment by Hunter-Bobeck** [comment]
> Are you using a plugin to make the emoji be images? It looks like your markup is different to what I see in my obsidian, so the plugin doesn't handle it

Oh I think Emoji Toolbar was doing that.
I turned it off and the images were no longer being done.

However, there is still an issue.
I tried turning off both Emoji Toolbar, and your Emoji Styling plugin. No more of an issue.
I tried turning on your Emoji Styling plugin then, all on its own. It's the culprit of this other issue. Here's the issue I'm seeing:
editing mode:
![image](https://user-images.githubusercontent.com/8333904/165840789-cb7581e9-916a-440f-96d6-191330c51475.png)
reading mode:
![image](https://user-images.githubusercontent.com/8333904/165840816-9983e602-1165-4159-a9ce-081f575e3fe9.png)

**Comment by Hunter-Bobeck** [comment]
So I didn't realize that Emoji Toolbar was setting all of my emoji to Twitter's versions of emoji in the reading mode. I would rather use the same emoji set in both reading mode and editing mode, so I decided to disable that feature of Emoji Toolbar. Sorry for that causing additional confusion as another interaction going on.

Furthermore, I ran into an issue with Dynamic Highlights where I was having trouble clicking on internal links, even when no code was put into Dynamic Highlights to have it highlight anything. I ran into enough of that issue and decided to uninstall Dynamic Highlights. (By the way, I posted that issue [here](https://github.com/nothingislost/obsidian-dynamic-highlights/issues/58) in case you are curious, but I am definitely not asking you to look into another plugin's issue!)

This leaves me with your awesome theme that, out of the box, paints gradients over emoji in headers, which I'd rather it not do. There's also your new plugin, 'Emoji Styling' (which for some reason adds all that unnecessary HTML you see in my last post). So your Emoji Styling doesn't seem to work for me, unfortunately.

While I could potentially combo your Emoji Styling with Dynamic Highlights, neither of them seem to work without an issue. So I'm not sure what to do right now.
